pkg/appgw/ingress_rules.go

```diff
@@ -18,31 +18,43 @@ func (c *appGwConfigBuilder) getListenersFromIngress(ingress *v1beta1.Ingress, e
 	listeners := make(map[listenerIdentifier]listenerAzConfig)
 
 	// if ingress has only backend configured
-	if ingress.Spec.Backend != nil && len(ingress.Spec.Rules) == 0 {
+	if ingress.Spec.Backend != nil && ingress.Spec.TLS == nil && len(ingress.Spec.Rules) == 0 {
+		glog.V(3).Infof("Only default backend defined for the ingress %s/%s", ingress.Namespace, ingress.Name)
 		return listeners
 	}
 
 	// process ingress rules with TLS and Waf policy
 	policy, _ := annotations.WAFPolicy(ingress)
+	if len(ingress.Spec.Rules) == 0 {
```

Comment:
Can this check be `ingress.Spec.Backend != nil` ?

Reply:
The line is checking when ingress is defined with default backend and TLS, in which case, TLS will be applied to the default backend

> Can this check be `ingress.Spec.Backend != nil` ?

The line is checking when ingress is defined with default backend and TLS, in which case, TLS will be applied to the default backend